core/frontend/src/types/autopilot/parameter.ts

```diff
@@ -0,0 +1,55 @@
+export default class Parameter {
+    name = ''
+
+    value = 0
+
+    readonly = false
+
+    description = ''
+
+    shortDescription = ''
+
+    options: {[key:number] : string} | undefined = undefined
+
+    bitmask: {[key:number] : string} | undefined = undefined
+
+    rebootRequired = false
+
+    id = 0
+
+    increment: number | undefined
+
+    constructor(name: string, id: number, value: number) {
+      this.name = name
+      this.value = value
+      this.id = id
+    }
+
+    setDescription(description: string): void {
+      this.description = description
+    }
+
+    setOptions(options: {[key:number] : string} | undefined): void {
+      this.options = options
+    }
+
+    setShortDescription(description: string): void {
+      this.shortDescription = description
+    }
+
+    setReadOnly(readonly: boolean): void {
+      this.readonly = readonly
+    }
+
+    setBitmask(bitmask: {[key:number] : string} | undefined): void {
+      this.bitmask = bitmask
+    }
+
+    setIncrement(increment: number | undefined): void {
+      this.increment = increment
+    }
+
+    setRebootRequired(required: boolean): void {
+      this.rebootRequired = required
+    }
```

Comment:
Lol, this seems like such a redundant design pattern. I take it javascript/typescript don't support direct assignment of instance attribute values?